### PR TITLE
Adjust platform specific code to recent changes

### LINKED ISSUE #1996: Lighting glitches when removing blocks
### Server Implementation

Paper

### Server Version

1.19.2

### Describe the bug

Paper build: git-Paper-245
I don't know if this is because of the latest paper build but when removing an area with WorldEdit lighting glitches appear:

Example of `//set 0` on the ground:

### To Reproduce

1. Select a region on the ground
2. Use //set 0

### Expected behaviour

There should not be any lighting glitches

### Screenshots / Videos

_No response_

### Error log (if applicable)

_No response_

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/426550cb049a4b36b747bc9abb874a96

### Fawe Version

FastAsyncWorldEdit version 2.4.9-SNAPSHOT-296;0d112b2

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit/ and the issue still persists.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by PierreSchwang**
## Overview
Fixes #1996

## Description
Paper moved some stuff to a new package and rewrote the chunk system. Big headache in our case. Should work. Tested a bit, but not completely, especially on non-paper builds (pre chunk rewrite). Possibly closes #1995 as well.
In total a big shitshow which is meant to break into thousands of pieces.

## Submitter Checklist

- [kind of] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: worldedit-bukkit/adapters/adapter-1_19/build.gradle.kts [hunk 1]
```diff
@@ -12,6 +12,6 @@ repositories {
 }
 
 dependencies {
-    paperDevBundle("1.19.2-R0.1-20220805.230830-1")
+    paperDevBundle("1.19.2-R0.1-SNAPSHOT")
```
Comment (NotMyFault) [inline worldedit-bukkit/adapters/adapter-1_19/build.gradle.kts:15]: Please pin the version, otherwise builds will fail, if more breaking changes are coming ahead mid-season, like this one, and nobody knows _where_ it's coming from.

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java [hunk 2]
```diff
@@ -106,6 +107,9 @@ public final class PaperweightPlatformAdapter extends NMSAdapter {
 
     private static final Field fieldRemove;
 
+    public static final boolean POST_CHUNK_REWRITE;
```
Comment (NotMyFault) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java:110]:
```suggestion
   static final boolean POST_CHUNK_REWRITE;
```
We can safely weaken the access modifier.

**Review by NotMyFault — COMMENTED** [review]
Works fine so far, based on a quick test

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightFaweAdapter.java [hunk 3]
```diff
@@ -674,4 +676,16 @@ public IBatchProcessor getTickingPostProcessor() {
         return new PaperweightPostProcessor();
     }
 
+    private boolean wasAccessibleSinceLastSave(ChunkHolder holder) {
+        if (!PaperLib.isPaper() || !PaperweightPlatformAdapter.POST_CHUNK_REWRITE) {
+            try {
+                return (boolean) holder.getClass().getDeclaredMethod("wasAccessibleSinceLastSave").invoke(holder);
```
Comment (SirYwell) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightFaweAdapter.java:682]: The method instance should be cached

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java [hunk 4]
```diff
@@ -591,9 +603,27 @@ static void removeBeacon(BlockEntity beacon, LevelChunk levelChunk) {
...
+            if (POST_CHUNK_REWRITE) {
+                try {
+                    //noinspection unchecked
+                    return (List<Entity>) PAPER_CHUNK_GEN_ALL_ENTITIES.invoke(chunk.level.getEntityLookup().getChunk(chunk.locX, chunk.locZ));
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            try {
+                EntityList entityList = (EntityList) LevelChunk.class.getDeclaredField("entities").get(chunk);
```
Comment (SirYwell) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java:615]: The field should be cached here, same below

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java [hunk 5]
```diff
@@ -591,9 +603,27 @@ static void removeBeacon(BlockEntity beacon, LevelChunk levelChunk) {
...
+                // fall through
+            }
+        }
+        try {
+            ((PersistentEntitySectionManager<Entity>) (ServerLevel.class.getDeclaredField("entityManager").get(chunk.level))).getEntities(chunk.getPos());
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            // fall through
         }
-        return chunk.level.entityManager.getEntities(chunk.getPos());
+        return List.of();
```
Comment (SirYwell) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/PaperweightPlatformAdapter.java:626]: Not sure if this is a good fallback, it should work on all platforms and ignoring all exceptions makes detecting bugs here very difficult.

**Review by SirYwell — DISMISSED** [review]

File: worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/regen/PaperweightRegen.java [hunk 6]
```diff
@@ -364,7 +364,7 @@ protected void cleanup() {
             Fawe.instance().getQueueHandler().sync(() -> {
                 try {
                     freshChunkProvider.close(false);
-                } catch (IOException e) {
+                } catch (Exception e) {
```
Comment (MattBDev) [inline worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/regen/PaperweightRegen.java:367]: Why was this changed to a more generic exception?

Comment (PierreSchwang) [reply worldedit-bukkit/adapters/adapter-1_19/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_19_R1/regen/PaperweightRegen.java:367]: Pre-Chunk Update the method throws an IOException, after the update, no exception is thrown anymore :/

**Comment by PierreSchwang** [comment]
Tested on Paper 1.19.2-256 (Post-Chunk-Update), Paper 1.19.2-142 (Pre-Chunk-Update) and Spigot 1.19.2 (latest build).
`//copy -e` & `//paste -e` and setting blocks for light updates

**Review by NotMyFault — APPROVED** [review]
Tested on the same build like @PierreSchwang, looks fine so far

**Review by SirYwell — APPROVED** [review]
